app/data_models/list_store.py

```diff
@@ -67,10 +67,10 @@ def __repr__(self):
     def first(self):
         try:
             return self.items[0]
-        except IndexError:
+        except IndexError as first_item_missing:
             raise IndexError(
                 f"unable to access first item in list, list '{self.name}' is empty"
-            )
+            ) from first_item_missing
```

Comment:
I prefer to keep the reference as `as e:`, `as ex:`, `as exc:` as that is common pattern in Python. I guess developer preference.